Kickstarter-iOS/DataSources/CommentRepliesDataSource.swift

```diff
@@ -234,7 +233,28 @@ internal final class CommentRepliesDataSource: ValueCellDataSource {
     return nil
   }
 
-  public func isCellInViewMoreRepliesSection(_ indexPath: IndexPath) -> Bool {
-    return indexPath.section == Section.viewMoreReplies.rawValue
+  /// When this function is called we set a `CommentViewMoreRepliesFailedCell` in the `viewMoreRepliesError` section of the data source.
+  private func showErrorState() {
+    self.set(
+      values: [()],
+      cellClass: CommentViewMoreRepliesFailedCell.self,
+      inSection: Section.viewMoreRepliesError.rawValue
+    )
+  }
+
+  /**
+   Returns `true` when  the `IndexPath`provided is from `Section.viewMoreReplies` or `Section.viewMoreRepliesError`.
+
+   - parameter indexPath: `IndexPath` object that we need the `Section` value from.
+
+   - returns: A  `Bool` which is only true is the section is either `viewMoreReplies` or `viewMoreRepliesError`.
+   */
+  public func sectionFor(_ indexPath: IndexPath) -> Bool {
```

Comment:
Replaced the `isCellInViewMoreRepliesSection` method with this one because we are now supporting two different types of cells in `didSelectRowAt`.

Reply:
The Section is internal to the data source so I didn't want to expose any of that in the view controller. Let me know what you think, I can change if it makes it easier for subsequent PRs.

Reply:
Best to do as much switching logic in the datasource as possible because we can cover that with tests.

Reply:
Cool, going to keep this as is.

Reply:
Could we name it `sectionForViewMoreReplies`?

Reply:
Thing is, it's checking viewMoreReplies and viewMoreRepliesError so I think that's a bit misleading.